File: BattleScreenJava/app/src/main/java/com/example/maxim_ignatiuc/battlescreen/MainActivity.java
package com.example.maxim_ignatiuc.battlescreen;


import android.app.Activity;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.TextView;
import android.view.animation.TranslateAnimation;


public class MainActivity extends Activity {

    boolean keepRun = true;
    int totalProgressTime = 100;
    private Handler progressBarbHandler = new Handler();
    private Thread t;
    private ProgressBar progress;

    TextView tv;
    RelativeLayout rl;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        progress = (ProgressBar) findViewById(R.id.progressBar1);
        progress.setProgress(totalProgressTime);






        rl = (RelativeLayout) findViewById(R.id.rl);
        tv = (TextView) findViewById(R.id.myTextView);




       final TranslateAnimation moveLefttoRight = new TranslateAnimation(100, 420, 0, 0);
        moveLefttoRight.setDuration(30);
        moveLefttoRight.setRepeatCount(1);
        moveLefttoRight.setRepeatMode(2);
        moveLefttoRight.setFillAfter(false);





        progress.setOnClickListener(new OnClickListener() {
            public void onClick(View view) {
                if (totalProgressTime >= 100) {
                    tv.startAnimation(moveLefttoRight);
                    progress.setProgress(0);
                    totalProgressTime = 0;
                }
            }
        });



        t = new Thread(new Runnable(){

            @Override
            public void run() {

                while (keepRun) {

                    if (totalProgressTime >= 100) {
                        progress.setProgress(100);
                    } else {
                        try {
                            totalProgressTime += 5;
                            Thread.sleep(80);
                        } catch (InterruptedException e) {
                            e.printStackTrace();
                        }
                        progressBarbHandler.post(new Runnable() {
                            public void run() {
                                progress.setProgress(totalProgressTime);
                            }
                        });

                    }
                }


            }

        });

t.start();

}

    @Override
    protected void onDestroy() {
        keepRun = false;
        try {
            t.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        super.onDestroy();
    }
}

File: BattleScreenRM/app/main_activity.rb
class MyRunnable
   def run
      while(@keepRun)
         if (@totalProgressTime >= 100) 
            @progress.setProgress(100)
         else
            @totalProgressTime = @totalProgressTime + 5;
            Thread.sleep(80)
         end       
                            
                        #} catch (InterruptedException e) {
                         #   e.printStackTrace();
                        #}

        @progressBarHandler.post(MyRunnable1)
      end
   end
end

class MyRunnable1
   def run
      @progress.setProgress(@totalProgressTime)
   end
end

#################################################################

class MainActivity < Android::App::Activity
 
   

  def onCreate(savedInstanceState)
    super
    
    @totalProgressTime = 100
    @keeprun = true
    @progressBarbHandler = Android::Os::Handler.new


    setContentView(resources.getIdentifier('activity_main', 'layout', 'com.yourcompany.battlescreenrm'))

   

    @progress = findViewById(resources.getIdentifier('progressBar1', 'id', 'com.yourcompany.battlescreenrm'))
    @progress.setProgress(@totalProgressTime)

    @rl =  findViewById(resources.getIdentifier('progressBar1', 'id', 'com.yourcompany.battlescreenrm'))
    @tv = findViewById(resources.getIdentifier('myTextView', 'id', 'com.yourcompany.battlescreenrm'))

    
    

    @moveLefttoRight = Android::View::Animation::TranslateAnimation.new(100, 200, 0, 0)
        @moveLefttoRight.setDuration(30)
        @moveLefttoRight.setRepeatCount(1);
        @moveLefttoRight.setRepeatMode(2);
        @moveLefttoRight.setFillAfter(false);
     
     @progress.setOnClickListener (self)    
    
  end


#@t = Java::Lang::Thread.new(MyRunnable)
#@t.start

  


# def onClick(view)
#     if(@totalProgressTime==100)
#        @totalProgressTime = 0      
#        @progress.setProgress(@totalProgressTime)
#        
#         @tv.startAnimation(@moveLefttoRight)
#        while @totalProgressTime <= 100 do
#           @totalProgressTime = @totalProgressTime + 1
#           @progress.setProgress(@totalProgressTime)
#        end
#     end  
# end


def onClick(view)
        @totalProgressTime = 0      
        @progress.setProgress(@totalProgressTime)
        @tv.startAnimation(@moveLefttoRight)
end




#def onDestroy
 #   @keepRun = false
     
  #  super
 # end




end




